hack/vendor.sh

```diff
@@ -1,28 +1,37 @@
 #!/usr/bin/env bash
 
-# This file is just wrapper around vndr (github.com/LK4D4/vndr) tool.
-# For updating dependencies you should change `vendor.conf` file in root of the
-# project. Please refer to https://github.com/LK4D4/vndr/blob/master/README.md for
-# vndr usage.
+# This file is just wrapper around 'go mod vendor' tool.
+# For updating dependencies you should change `vendor.mod` file in root of the
+# project.
 
 set -e
 
-if ! hash vndr; then
-	echo "Please install vndr with \"go get github.com/LK4D4/vndr\" and put it in your \$GOPATH"
-	exit 1
-fi
+cat > go.mod << EOF
+module github.com/moby/moby
+
+go 1.17
+EOF
+
+cat > hack/make/.resources-windows/go.mod << EOF
+module github.com/docker/docker/autogen/winresources/dockerd
+
+go 1.17
+EOF
 
 if [ $# -eq 0 ] || [ "$1" = "archive/tar" ]; then
 	echo "update vendored copy of archive/tar"
 	: "${GO_VERSION:=$(awk -F '[ =]' '$1 == "ARG" && $2 == "GO_VERSION" { print $3; exit }' ./Dockerfile)}"
-	rm -rf vendor/archive
-	mkdir -p ./vendor/archive/tar
+	rm -rf vendor/archive/tar internal/archive/tar
+	mkdir -p internal/archive/tar
 	echo "downloading: https://golang.org/dl/go${GO_VERSION%.0}.src.tar.gz"
 	curl -fsSL "https://golang.org/dl/go${GO_VERSION%.0}.src.tar.gz" \
-		| tar --extract --gzip --directory=vendor/archive/tar --strip-components=4 go/src/archive/tar
-	patch --strip=4 --directory=vendor/archive/tar --input="$PWD/patches/0001-archive-tar-do-not-populate-user-group-names.patch"
+		| tar --extract --gzip --directory=internal/archive/tar --strip-components=4 go/src/archive/tar
+	patch --strip=4 --directory=internal/archive/tar --input="$PWD/patches/0001-archive-tar-do-not-populate-user-group-names.patch"
 fi
 
 if [ $# -eq 0 ] || [ "$1" != "archive/tar" ]; then
-	vndr -whitelist='^archive[/\\]tar' "$@"
+	go mod vendor -modfile vendor.mod
```

Comment:
Not 100% sure; do we need a `go mod tidy` before `go mod vendor` ?

Looks like we need to set `GO111MODULES=auto` here. When I try to run it inside the "dev" container (`make BIND_DIR=. shell`), it complains that modules are disabled;

```
hack/vendor.sh
update vendored copy of archive/tar
downloading: https://golang.org/dl/go1.17.4.src.tar.gz
patching file stat_unix.go
Hunk #1 succeeded at 9 (offset 1 line).
Hunk #2 succeeded at 17 (offset 1 line).
Hunk #3 succeeded at 25 (offset 1 line).
go: modules disabled by GO111MODULE=off; see 'go help modules'
```

Reply:
I think `go mod tidy` should be called explicitly, it has too much of a side-effect :)

Reply:
> Looks like we need to set `GO111MODULES=auto` here. When I try to run it inside the "dev" container (`make BIND_DIR=. shell`), it complains that modules are disabled;

fixed 👍

Reply:
This seems to happen if you run it without `-modfile vendor.mod`, but with that flag these errors went away when I check last time.

Reply:
Per @crazy-max's suggestions I've just added `go mod tidy` to `hack/validate/vendor`, hope that addresses your concern.